pkg/pipeline/write/write_loki.go

```diff
@@ -271,7 +266,9 @@ func NewWriteLoki(params config.StageParam) (*Loki, error) {
 		}
 	}
 
+	// TODO / FIXME / FIGUREOUT: seems like we have 2 input channels for Loki? (this one, and see also pipeline_builder.go / getStageNode / StageWrite)
```

Comment:
This channel seems to act as a buffer to avoid blocking the previous pipeline. Probably we'd need to verify that removing the `in` channel wouldn't affect performance.

When we are able to use Go 1.18, maybe we could upgrade to an [upper-upperstream version of the go-pipes](https://github.com/mariomac/pipes) library that allows defining the buffer length between pipeline stages.